src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -214,6 +202,34 @@ void mf::XWaylandServer::connect_wm_to_xwayland()
 
     if (xwayland_startup_timed_out)
     {
-        mir::fatal_error("XWayland failed to start");
+        BOOST_THROW_EXCEPTION(std::runtime_error("XWayland server failed to start"));
+    }
+
+    return client;
+}
+}
+
+mf::XWaylandServer::XWaylandServer(
+    std::shared_ptr<WaylandConnector> const& wayland_connector,
+    XWaylandSpawner const& spawner,
+    std::string const& xwayland_path)
+    : xwayland_process{fork_xwayland_process(spawner, xwayland_path)},
+      wayland_client{connect_xwayland_wl_client(wayland_connector, xwayland_process.wayland_fd)}
+{
+}
+
+mf::XWaylandServer::~XWaylandServer()
+{
+    mir::log_info("Deiniting xwayland server");
+
+    // Terminate any running xservers
+    if (kill(xwayland_process.pid, SIGTERM) == 0)
+    {
+        std::this_thread::sleep_for(100ms);// After 100ms...
```

Comment:
We could do better than sleeping by waiting for the child to exit, but maybe that can wait for a `Process` abstraction…

Reply:
I don't know that we need a new abstraction, but that's at least a task for another PR